hooks/README.md

```diff
@@ -0,0 +1,98 @@
+
+Remember your hook labels events, not images, or videos. One event may contain mutiple images and multiple videos. The same video can span multiple consecutive events. At this time we only send to your hook preprocessed images. Also, you only label events that Camio has alreday pre-filtered for you as interesting.
+
+## Installing the example hook code
+
+The attached code include an installation script for Debian/Ubuntu 
+
+   hook-install.sh
+
+and an example of a hook:
+
+   hook-example.py
+
+You can run the install with 
+
+   ./hook-install.sh
+
+and it will install the required dependencies and start two processes:
+
+- a web server that received the hook calls and enqueues requests
+- a background process that handles the requests and posts back labels to Camio
+
+The queue is implemented as a mongodb database collection.
+
+## The example hook
+
+Our hook-example.py depends on bottle (0.13), gunicorn, requests, PIL, and pymongo.
+
+The hook-example.py code is build on bottle.py 0.13 and intended to work any WSGI web 
+server. We recommend gunicorn (a prefork WSGI server) and our install script assumes it.
+Once you have all the dependencies installed you can start the web server with:
+
+    nohup gunicorn -w 2 app:hook-example -b 0.0.0.0:8000 > /tmp/gunicorn.log &
+
+Here 8000 is the post to be used and 0.0.0.0 refers to (any IP address). 
+-w 2 requests two web server workers and /tmp/gunicorn.log is the location of the logfile.
+
+The server will expose three endpoints:
+
+- http://{your domain}:8000/tasks/ (GET) which you can call to check the service is running
+- http://{your domain}:8000/tasks/{api_key} (POST) which will you have to register with Camio and be called by Camio to post event images
+- http://{your domain}:8000/tasks/{api_key} (GET) which you can call to obtain a list of pending tasks
+
+The {api_key} is your own API key and you can make it up to be whatever you want. It has to match the API_KEY global variable in the example code. The purpose of the API_KEY is to allow Camio to gain access to your hook and prevent un-authroized access.
+
+Long with the web server you must start the background process:
+
+    nohup python hook-example.py > /tmp/taskqueue.log &
+
+The background process retrieves pending tasks collected by the hook and posts computed labels back to Camio. They will be added to the originating event.
+
+The hook-example.py depends on the following function:
+
+    def compute_labels(images):
+        labels = []
+        for image in images:
+            ...
+        labels = ['cat','dog','mouse']
+        return labels
+
+This does nothing more than returning the same three labels (cat, dog, mouse) for each 
+event but you can edit it to user your favorite ML or NN tool such as OpenCV or Tensorflow,
+to perform Object Detection and compute labels from the images.
+
+## registering the hook
+
+Once a hook is created you must tell Camio about it. You must all tell Camio which events
```

Comment:
A Hook is by definition a hook into the Camio pipeline. So for terminology, the user creates a labeling server and then registers it as a Camio Hook with the POST to camio.com.